flink-streaming-scala/src/main/scala/org/apache/flink/streaming/api/scala/ConnectedStreams.scala

```diff
@@ -300,19 +300,22 @@ class ConnectedStreams[IN1, IN2](javaStream: JavaCStream[IN1, IN2]) {
    * @param fun2 The second stream's key function
    * @return The key-grouped connected streams
    */
-  def keyBy[K1: TypeInformation, K2: TypeInformation](fun1: IN1 => K1, fun2: IN2 => K2):
+  def keyBy[KEY: TypeInformation](fun1: IN1 => KEY, fun2: IN2 => KEY):
       ConnectedStreams[IN1, IN2] = {
 
-    val keyType1 = implicitly[TypeInformation[K1]]
-    val keyType2 = implicitly[TypeInformation[K2]]
-    
+    val keyType = implicitly[TypeInformation[KEY]]
+
     val cleanFun1 = clean(fun1)
     val cleanFun2 = clean(fun2)
-    
-    val keyExtractor1 = new KeySelectorWithType[IN1, K1](cleanFun1, keyType1)
-    val keyExtractor2 = new KeySelectorWithType[IN2, K2](cleanFun2, keyType2)
-    
-    asScalaStream(javaStream.keyBy(keyExtractor1, keyExtractor2))
+
+    val keyExtractor1 = new KeySelector[IN1, KEY] {
```

Comment:
nit: I like the simplicity of the code but I think stacktraces are harder to parse with anonymous classes thrown in the mix. Just a thought, though.

Reply:
I reverted the internal class. 

My main goal here was to pass the key type explicitly rather than through the `ResultTypeQueryable`, which requires running `TypeExtractor`.